### PR TITLE
View Pledge [UI]

### PR DESCRIPTION
**Pull request by Rcureton**
# What ❓
- Added the logic to show the proper `button string` and `button color` for `View your pledge, Manage, and Back this project.`

# How to QA? 🤔
- Go to Projects you've backed that have completed and when you click the project the reward button should read `View your pledge`

# Story 📖
[Move View my pledge into Rewards container](https://trello.com/c/Klxmcsgd/1280-move-view-my-pledge-into-rewards-container)

# See 👀

### CODE REVIEW

**Comment by eoji** [comment]
Question: I thought the Trello card was just for `View your pledge` but I also see `Manage my pledge`.  I also don't see `View Rewards` so what Trello cards is this PR for?

**Comment by Rcureton** [comment]
It is for `View my pledge` but I added the logic for `manage pledge` since it existed together previously. So I figured it would be easier for the next person that takes that story to implement.

Should I remove it to keep the story within scope?

**Comment by eoji** [comment]
Yes, please. Because "Manage my pledge" is missing the reward details.

**Comment by Rcureton** [comment]
I removed it 👍🏾

File: app/src/main/java/com/kickstarter/viewmodels/ProjectViewModel.kt [hunk 1]
```diff
@@ -79,6 +77,12 @@ interface ProjectViewModel {
         /** Emits when we should set the Y position of the rewards container. */
         fun setInitialRewardsContainerY(): Observable<Void>
 
+        /** Emits the color for the reward button based on (View, Manage, or Back this project). */
+        fun setRewardButtonColor(): Observable<Int>
```
Comment (eoji) [inline app/src/main/java/com/kickstarter/viewmodels/ProjectViewModel.kt]: let's rename this to `rewardsButtonColor`

File: app/src/main/java/com/kickstarter/viewmodels/ProjectViewModel.kt [hunk 2]
```diff
@@ -79,6 +77,12 @@ interface ProjectViewModel {
...
+
+        /** Emits the proper string for the reward button. */
+        fun setRewardButtonString(): Observable<Int>
```
Comment (eoji) [inline app/src/main/java/com/kickstarter/viewmodels/ProjectViewModel.kt]: let's rename this to `rewardsButtonText`

File: app/src/main/java/com/kickstarter/viewmodels/ProjectViewModel.kt [hunk 3]
```diff
@@ -79,6 +77,12 @@ interface ProjectViewModel {
         /** Emits when we should set the Y position of the rewards container. */
         fun setInitialRewardsContainerY(): Observable<Void>
 
+        /** Emits the color for the reward button based on (View, Manage, or Back this project). */
+        fun setRewardButtonColor(): Observable<Int>
+
+        /** Emits the proper string for the reward button. */
```
Comment (eoji) [inline app/src/main/java/com/kickstarter/viewmodels/ProjectViewModel.kt]: It's not emitting a string, it's emitting a string resource ID.

File: app/src/main/java/com/kickstarter/viewmodels/ProjectViewModel.kt [hunk 4]
```diff
@@ -79,6 +77,12 @@ interface ProjectViewModel {
         /** Emits when we should set the Y position of the rewards container. */
         fun setInitialRewardsContainerY(): Observable<Void>
 
+        /** Emits the color for the reward button based on (View, Manage, or Back this project). */
```
Comment (eoji) [inline app/src/main/java/com/kickstarter/viewmodels/ProjectViewModel.kt]: It's not emitting a string, it's emitting a color resource ID.

File: app/src/main/java/com/kickstarter/viewmodels/ProjectViewModel.kt [hunk 5]
```diff
@@ -318,6 +324,18 @@ interface ProjectViewModel {
                     .compose(bindToLifecycle())
                     .subscribe { this.setActionButtonId.onNext(it) }
 
+            currentProject
+                    .map { getRewardButtonString(it) }
+                    .take(1)
```
Comment (eoji) [inline app/src/main/java/com/kickstarter/viewmodels/ProjectViewModel.kt]: Is there a reason why you're doing `take(1)` if the project ever emits after the user backs it, the button will never update.

File: app/src/main/java/com/kickstarter/viewmodels/ProjectViewModel.kt [hunk 6]
```diff
@@ -318,6 +324,18 @@ interface ProjectViewModel {
...
+                    .compose(bindToLifecycle())
+                    .subscribe { this.setRewardButtonString.onNext(it) }
+
+            currentProject
+                    .map { getRewardButtonColor(it) }
+                    .take(1)
```
Comment (eoji) [inline app/src/main/java/com/kickstarter/viewmodels/ProjectViewModel.kt]: Same here

File: app/src/main/java/com/kickstarter/ui/activities/ProjectActivity.kt [hunk 7]
```diff
@@ -84,6 +84,16 @@ class ProjectActivity : BaseActivity<ProjectViewModel.ViewModel>() {
                     }
                 }
 
+        this.viewModel.outputs.setRewardButtonColor()
+                .compose(bindToLifecycle())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe { it?.let { native_back_this_project_button.setBackgroundColor(getColor(it)) } }
```
Comment (eoji) [inline app/src/main/java/com/kickstarter/ui/activities/ProjectActivity.kt]: Screen Shot 2019-06-10 at 12 11 36 PM

This code has a warning. The inner or outer `it` should be renamed.

File: app/src/main/java/com/kickstarter/ui/activities/ProjectActivity.kt [hunk 8]
```diff
@@ -84,6 +84,16 @@ class ProjectActivity : BaseActivity<ProjectViewModel.ViewModel>() {
...
+
+        this.viewModel.outputs.setRewardButtonString()
+                .compose(bindToLifecycle())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe { it?.let { native_back_this_project_button.text = getString(it) } }
```
Comment (eoji) [inline app/src/main/java/com/kickstarter/ui/activities/ProjectActivity.kt]: Same thing here.

File: app/src/test/java/com/kickstarter/viewmodels/ProjectViewModelTest.kt [hunk 9]
```diff
@@ -27,6 +27,8 @@ class ProjectViewModelTest : KSRobolectricTestCase() {
     private val savedTest = TestSubscriber<Boolean>()
     private val setActionButtonId = TestSubscriber<Int>()
     private val setInitialRewardsContainerY = TestSubscriber<Void>()
+    private val setRewardButtonColor = TestSubscriber<Int>()
```
Comment (eoji) [inline app/src/test/java/com/kickstarter/viewmodels/ProjectViewModelTest.kt]: Plz don't forget to rename it in the tests.

**Comment by eoji** [comment]
[Move View my pledge into Rewards container](https://trello.com/c/Klxmcsgd/1280-move-view-my-pledge-into-rewards-container)

**Pull request merged**